portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -74,19 +79,39 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
         BlogMessage message = new BlogMessage(messageId, tags, comment, image, sender, messageReplies, timestamp);
         messages.add(message);
       }
+      
```

Comment:
Normally, linking this into the other parts of the code would be done in a separate PR. For this case it would be for two reasons: to make the PR smaller and more focused on a single goal (implementation), and because this code isn't tested yet. This would be a perfect chance to practice branching a new change off of your current change and migrating this code to that new branch.